bitnami/prometheus-operator/templates/alertmanager/alertmanager.yaml

```
@@ -52,8 +52,19 @@ spec:
         {{- include "prometheus-operator.alertmanager.storageClass" . }}
   {{- end }}
   {{- end }}
-  {{- if .Values.alertmanager.podMetadata }}
-  podMetadata: {{- include "prometheus-operator.tplValue" (dict "value" .Values.alertmanager.podMetadata "context" $) | nindent 4 }}
+  {{- if or .Values.alertmanager.podMetadata.labels .Values.alertmanager.podMetadata.annotations (eq .Values.alertmanager.podAntiAffinity "soft") (eq .Values.alertmanager.podAntiAffinity "hard") }}
+  podMetadata:
+    labels:
+    {{- if .Values.alertmanager.podMetadata.labels }}
+    {{- include "prometheus-operator.tplValue" (dict "value" .Values.alertmanager.podMetadata.labels "context" $) | nindent 6 }}
+    {{- end }}
+    {{- if or (eq .Values.alertmanager.podAntiAffinity "soft") (eq .Values.alertmanager.podAntiAffinity "hard") }}
+    {{- include "prometheus-operator.alertmanager.matchLabels" . | nindent 6 }}
+    {{- end }}
```

Do not define labels when only `alertmanager.podMetadata.annotations` are defined:
```suggestion
    {{- if or .Values.alertmanager.podMetadata.labels (eq .Values.alertmanager.podAntiAffinity "soft") (eq .Values.alertmanager.podAntiAffinity "hard") }}
    labels:
      {{- if .Values.alertmanager.podMetadata.labels }}
      {{- include "prometheus-operator.tplValue" (dict "value" .Values.alertmanager.podMetadata.labels "context" $) | nindent 6 }}
      {{- end }}
      {{- if or (eq .Values.alertmanager.podAntiAffinity "soft") (eq .Values.alertmanager.podAntiAffinity "hard") }}
      {{- include "prometheus-operator.alertmanager.matchLabels" . | nindent 6 }}
      {{- end }}
    {{- end }}
```